### ISSUE TITLE
Problem with JUnit-4.12-beta-1

### ISSUE THREAD

**Issue by Flic182** [valfirst/jbehave-junit-runner#70]
When trying to run a JUnit test in Eclipse with the @RunWith(JUnitReportingRunner.class) annotation, there seems to be a problem caused by using JUnit-4.12-beta-1.  The problem is not evident with JUnit-4.11.  Here is the stack trace:

java.lang.RuntimeException: java.lang.NullPointerException
at de.codecentric.jbehave.junit.monitoring.JUnitReportingRunner.run(JUnitReportingRunner.java:80)
at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.lang.NullPointerException
at de.codecentric.jbehave.junit.monitoring.JUnitScenarioReporter.afterStory(JUnitScenarioReporter.java:114)
at org.jbehave.core.reporters.DelegatingStoryReporter.afterStory(DelegatingStoryReporter.java:49)
at org.jbehave.core.reporters.ConcurrentStoryReporter.afterStory(ConcurrentStoryReporter.java:120)
at org.jbehave.core.embedder.PerformableTree.performBeforeOrAfterStories(PerformableTree.java:399)
at org.jbehave.core.embedder.StoryManager.performStories(StoryManager.java:102)
at org.jbehave.core.embedder.StoryManager.runStories(StoryManager.java:93)
at org.jbehave.core.embedder.StoryManager.runStoriesAsPaths(StoryManager.java:74)
at org.jbehave.core.embedder.Embedder.runStoriesAsPaths(Embedder.java:204)
at de.codecentric.jbehave.junit.monitoring.JUnitReportingRunner.run(JUnitReportingRunner.java:78)
... 6 more

A sample program and test is given on Stack Overflow at this link:  

http://stackoverflow.com/questions/25589963/jbehave-junit-runner-throwing-nullpointerexception/25646847#25646847

Hopefully this is merely a symptom of the beta-status of JUnit-4.12, but I am raising this issue as per Andreas' request on the Stack Overflow question.

Kind regards,
Flic

**Comment by sheldonh** [comment]
Nope. Problem persists with JUnit-4.12 release.

**Comment by skurlow** [comment]
Yep I see the same issue with the 4.12 release on IntelliJ 13.1.2.

I haven't investigated much but I can see JUnit 4.12 doesn't recognise or is not informed by the jbehave junit runner that there are any stories, seems like it thinks there aren't any story files.

**Comment by soezen** [comment]
the problem lies with 
JunitReportingRunner:line 171
private void initRootDescription() {
 rootDescription = Description .createSuiteDescription(configurableEmbedder.getClass()); rootDescription.getChildren().addAll(storyDescriptions); 
}

rootDescription.getChildren returns a new array list of children, so doing addAll does not actually add the children to the actual list of Description
Description class does offer a method addChild, using this method would solve this issue

**Comment by britter** [comment]
@soezen would you like to create a PR for this?

**Comment by soezen** [comment]
done

**Comment by gmatheu** [comment]
Are you planning to release this fix (1.2.0) any time soon?

**Comment by britter** [comment]
@gmatheu yes, we're planning to provide a new release next week.

**Comment by piperfe** [comment]
awesome !! My team needs this feature !!

**Comment by mvitiuk** [comment]
Hello, I'm using 4.12 version and experiencing similar issue when trying to run scenario with Examples.

I have scenarios:
Scenario1: without examples
Scenario2: with examples
Scenario3: without examples
- When I run all scenarios everything fine
- When run only Scenario2 - NullPointerException (logs below)
- When run Scenario 1 and Scenario2 - everything fine
- When Scenario2 on the top _like Scenario2: with examples, Scenario1: without examples, Scenario3: without examples_\- everything fine
- When run Scenario2 and Scenario3 - NullPointerException (logs below) and Scenario3 not performed

Also strange behaviour, when Scenario2 on the top of the story, and I have some other scenarios with examples - run scenario one by one using meta filters works just fine. 

Could somebody guid me, should I post new  issue or this is the right place?  

``` java.lang.NullPointerException
    at de.codecentric.jbehave.junit.monitoring.JUnitScenarioReporter.example(JUnitScenarioReporter.java:240)
    at org.jbehave.core.reporters.DelegatingStoryReporter.example(DelegatingStoryReporter.java:91)
    at org.jbehave.core.reporters.ConcurrentStoryReporter.example(ConcurrentStoryReporter.java:212)
    at org.jbehave.core.embedder.PerformableTree$ExamplePerformableScenario.perform(PerformableTree.java:950)
    at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:849)
    at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:781)
    at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:758)
    at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:390)
    at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:364)
    at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:291)
    at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:265)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
```

**Comment by FedorRomanov** [comment]
any plans to integrate the fix into JUnit 4.12 ? it still throws NPE
```
Caused by: java.lang.NullPointerException
	at de.codecentric.jbehave.junit.monitoring.JUnitScenarioReporter.afterStory(JUnitScenarioReporter.java:108)
	at org.jbehave.core.reporters.DelegatingStoryReporter.afterStory(DelegatingStoryReporter.java:49)
```

**Comment by valfirst** [comment]
`jbehave-junit-runner` is based on JUnit 4.12 now. Use new release:
```xml
<dependency>
    <groupId>com.github.valfirst</groupId>
    <artifactId>jbehave-junit-runner</artifactId>
    <version>2.0.0</version>
</dependency>
```
If the issue is still reproduced, please report here or open a new issue
